### PR TITLE
[posix] add a flag to turn posix multicast routing feature on/off

### PR DESCRIPTION
**Pull request by sunytt**
This PR adds a separate flag to be able to turn the posix part of multicast routing implementation off/on.

The new OPENTHREAD_POSIX_CONFIG_BACKBONE_ROUTER_MULTICAST_ROUTING_ENABLE flag guards the feature of posix multicast routing using MFC(multicast forwarding cache) in MRT(multicast routing table).

The existing OPENTHREAD_CONFIG_BACKBONE_ROUTER_MULTICAST_ROUTING_ENABLE continues to guard the core multicast routing feature, for example handling the MLR.req message and the multicast listeners table.

In a system where it has its own multicast routing support for packet forwarding between Thread network and AIL, it should set OPENTHREAD_POSIX_CONFIG_BACKBONE_ROUTER_MULTICAST_ROUTING_ENABLE to 0.

### CODE REVIEW

**Review by wgtdkp — APPROVED** [review]
LGTM :+1:

File: src/posix/platform/openthread-posix-config.h [hunk 1]
```diff
@@ -125,6 +125,17 @@
 #define OPENTHREAD_POSIX_CONFIG_MAX_MULTICAST_FORWARDING_CACHE_TABLE (OPENTHREAD_CONFIG_MAX_MULTICAST_LISTENERS * 10)
 #endif
 
+/**
```
Comment (superwhd) [inline src/posix/platform/openthread-posix-config.h:128]: Should we also add a check that 
 when `OPENTHREAD_POSIX_CONFIG_BACKBONE_ROUTER_MULTICAST_ROUTING_ENABLE` is true,  `OPENTHREAD_CONFIG_BACKBONE_ROUTER_MULTICAST_ROUTING_ENABLE` must be true as well?

Comment (sunytt) [reply src/posix/platform/openthread-posix-config.h:128]: That'd make sense, done.

File: src/posix/platform/backbone.cpp [hunk 2]
```diff
@@ -42,7 +42,7 @@
 
 char         gBackboneNetifName[IFNAMSIZ] = "";
 unsigned int gBackboneNetifIndex          = 0;
-#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_MULTICAST_ROUTING_ENABLE
+#if OPENTHREAD_POSIX_CONFIG_BACKBONE_ROUTER_MULTICAST_ROUTING_ENABLE
```
Comment (jwhui) [inline src/posix/platform/backbone.cpp:45]: Should we include in the name something more specific to MFC / MRT? In the future, do we envision supporting multiple different ways of supporting multicast routing on posix platform?

Comment (sunytt) [reply src/posix/platform/backbone.cpp:45]: Currently only linux is supported for posix multicast routing, the implementation is guarded by `#if __linux__`. Other posix systems may support MFCs differently(e.g. netBSD), or not support MFCs. Some parts of the MulticastRoutingManager could possibly be reused by different implementations, e.g entry of SetUp(), TearDown(), HandleStateChange().

I think we could keep this flag general for posix systems and when adding support for other posix systems use the OS macros to guard the implementations. Thoughts?

**Review by abtink — APPROVED** [review]
LGTM. 👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**